libnm-core/nm-setting-wpan.c

```diff
@@ -26,22 +26,34 @@
 #include "nm-setting-private.h"
 #include "nm-utils-private.h"
 
-/* Ideally we'll be able to get this from a public header. */
+/* Ideally we'll be able to get these from a public header. */
 #ifndef IEEE802154_ADDR_LEN
 #define IEEE802154_ADDR_LEN 8
 #endif
 
+#ifndef IEEE802154_MAX_PAGE
+#define IEEE802154_MAX_PAGE 31
+#endif
+
+#ifndef IEEE802154_MAX_CHANNEL
+#define IEEE802154_MAX_CHANNEL 26
+#endif
+
 enum {
 	PROP_0,
 	PROP_MAC_ADDRESS,
 	PROP_PAN_ID,
 	PROP_SHORT_ADDRESS,
+	PROP_PAGE,
+	PROP_CHANNEL,
 };
 
 typedef struct {
 	char *mac_address;
 	guint16 pan_id;
 	guint16 short_address;
+	gint16 page;
+	gint16 channel;
```

Comment:
`gint16` is larger that the range of the property below (which is restricted to `-1` ... `IEEE802154_MAX_*`.

Below, I suggested, that the range of the GObject property should not be restricted.

either way, choose a int type that fits the size of the range from the GObject property. As it is now, gint8 would suffice. If the range is G_MININT32/G_MAXINT32, it would need gint32.